hail/src/main/scala/is/hail/expr/ir/AbstractMatrixTableSpec.scala

```diff
@@ -27,6 +27,9 @@ object RelationalSpec {
     new MatrixTypeSerializer
 
   def readMetadata(fs: FS, path: String): JValue = {
+    if (!fs.exists(path)) {
```

Comment:
can we move this into the `!isDir` path so we don't end up with two file system calls on the fast path?

Reply:
Good call.